### PR TITLE
ISD-249 add a juju action to show available variables for environment config option

### PR DESCRIPTION
**Pull request by gregory-schiano**

### CODE REVIEW

File: .woke.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Ignore ingress charm library as it uses this non compliant terminology: whitelist.
+ignore_files:
+ - lib/charms/nginx_ingress_integrator/v0/ingress.py
```
Comment (mthaddon) [inline .woke.yaml:3]: Can I suggest the same changes here as in https://github.com/canonical/discourse-k8s-operator/pull/10 ?

File: actions.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+show-environment-context:
+  description: List variables available in StoredState
```
Comment (mthaddon) [inline actions.yaml:2]: I would suggest we describe this in terms that make sense for the operator. Maybe something like "List variables available for use in the `environment` config option"? And maybe we should also add a reference to this action in the description for that config option.

File: .github/workflows/integration_test.yaml [hunk 3]
```diff
@@ -3,6 +3,9 @@ name: Integration tests
 on:
   pull_request:
 
+env:
```
Comment (arturo-seijas) [inline .github/workflows/integration_test.yaml:6]: Why do you need this here?

File: .woke.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+ignore_files:
+  # Ignore ingress charm library as it uses non compliant terminology:
+  # whitelist.
+  - lib/charms/nginx_ingress_integrator/v0/ingress.py
+rules:
+  # Ignore "master" - the database relation event received from the library
+  # uses this terminology.
+  - name: master
```
Comment (arturo-seijas) [inline .woke.yaml:8]: Do we need to include master in the whitelist?

File: src/charm.py [hunk 5]
```diff
@@ -181,6 +184,15 @@ def _on_gunicorn_pebble_ready(self, event: ops.framework.EventBase) -> None:
 
         self._configure_workload(event)
 
+    def _on_show_environment_context_action(self, event: ops.charm.ActionEvent) -> None:
+        """Handle event for show-environment-context action"""
+        logger.info("Action launched")
```
Comment (weiiwang01) [inline src/charm.py:189]: Maybe include the name of the action in log?

File: .woke.yaml (hunk 4 above)
Comment (gregory-schiano) [reply .woke.yaml:8]: Yes due to postgres

**Pull request merged**